### ISSUE TITLE
Service does not start on Ubuntu 14.04

### ISSUE THREAD

**Issue by jopasserat** [gsick/puppet-jetty#2]
It seems that commit 053f2595e9d88abbde74a27c9f8c2c07f0cd30ec introduces a regression that prevents the service to start on a fresh Ubuntu 14.04.

**Comment by gsick** [comment]
This commit has nothing to do, the default init script from jetty dist doesn't respect the linux spec. it should exit with 0 for start cmd when the process is already running.
I tried and tested on docker/ubuntu 14 and it works.

What is the error on your side?

**Comment by jopasserat** [comment]
This is what it returns in my case:

```
Error: Could not start Service[jetty]: Execution of '/etc/init.d/jetty start' returned 1: 
Error: /Stage[main]/Jetty/Service[jetty]/ensure: change from stopped to running failed: Could not    start Service[jetty]: Execution of '/etc/init.d/jetty start' returned 1:
```

and the service if launched manually (service jetty start) from the VM gives:

```
root@testCloud1:/etc/puppet/modules/jetty# service jetty restart
  Stopping Jetty: start-stop-daemon: warning: failed to kill 9559: No such process
  1 pids were not killed
  No process in pidfile '/var/run/jetty.pid' found running; none killed.
  OK
  Starting Jetty: FAILED Sat Jan 10 21:18:16 GMT 2015
```

on the other hand, using the command line that the service is supposed to run, launches is fine (regardless of the current directory, here `/tmp`)

```
root@testCloud1:/tmp# su jetty -c "/usr/bin/java -Djetty.home=/opt/jetty -Djetty.base=/opt/jetty -Djava.io.tmpdir=/tmp -jar /opt/jetty/start.jar jetty-started.xml &"
  root@testCloud1:/tmp# 2015-01-09 17:45:56.868:INFO::main: Logging initialized @397ms
  2015-01-09 17:45:57.167:INFO:oejs.Server:main: jetty-9.1.3.v20140225
  2015-01-09 17:45:57.201:INFO:oejdp.ScanningAppProvider:main: Deployment monitor [file:/opt/jetty-distribution-9.1.3.v20140225/webapps/] at interval 1
  2015-01-09 17:45:57.222:INFO:oejs.ServerConnector:main: Started   ServerConnector@6b2d2436{HTTP/1.1}{0.0.0.0:8080}
  2015-01-09 17:45:57.223:INFO:oejs.Server:main: Started @756ms
```

My fork that reverts the commit I mention provisions the machine fine and the service starts.

**Comment by gsick** [comment]
I have no clue.

I tried on ec2 - ubuntu 14.04. Everything looks fine

ubuntu@ip-xxxxx:/etc/puppet$ puppet --version
3.4.3
ubuntu@ip-xxxxx:/etc/puppet$ sudo puppet apply manifests/init.pp 
Notice: Compiled catalog for ip-xxxxx.ec2.internal in environment production in 1.14 seconds
Notice: /Stage[main]/Java/Package[java]/ensure: ensure changed 'purged' to 'present'
Notice: /Stage[main]/Jetty/Exec[download jetty]/returns: executed successfully
Notice: /Stage[main]/Jetty/File[jetty default]/ensure: created
........
Notice: /Stage[main]/Jetty/File[jetty home]/ensure: created
Notice: /Stage[main]/Jetty/File[jetty demo]/ensure: removed
Notice: /Stage[main]/Jetty/File[jetty init]/ensure: defined content as '{md5}71da8450d25685e584aaa02d5779792e'
Notice: /Stage[main]/Jetty/Service[jetty]/ensure: ensure changed 'stopped' to 'running'
Notice: Finished catalog run in 107.79 seconds

ubuntu@ip-xxxxx:/etc/puppet$ sudo service jetty status
Checking arguments to Jetty: 
START_INI      =  /opt/jetty/start.ini
JETTY_HOME     =  /opt/jetty
JETTY_BASE     =  /opt/jetty
JETTY_CONF     =  /opt/jetty/etc/jetty.conf
JETTY_PID      =  /var/run/jetty.pid
JETTY_START    =  /opt/jetty/start.jar
JETTY_LOGS     =  /var/log/jetty
JETTY_STATE    =  /opt/jetty/jetty.state
CLASSPATH      =  
JAVA           =  /usr/bin/java
JAVA_OPTIONS   =   -Djetty.home=/opt/jetty -Djetty.base=/opt/jetty -Djava.io.tmpdir=/tmp
JETTY_ARGS     =   jetty-started.xml
RUN_CMD        =  /usr/bin/java -Djetty.home=/opt/jetty -Djetty.base=/opt/jetty -Djava.io.tmpdir=/tmp -jar /opt/jetty/start.jar jetty-started.xml

Jetty running pid=13399
ubuntu@ip-xxxxx:/etc/puppet$ sudo service jetty restart
Stopping Jetty: OK
Starting Jetty: OK Mon Jan 12 17:16:27 UTC 2015

Did you try on a fresh server? Are you sure you don't have a jetty already running?

**Comment by jopasserat** [comment]
Yeah I've just destroyed and recreated my VM. It might be because I'm using a slightly modified Ubuntu customized by my ops team. But as far as I know they only perform modifications related to security, nothing that should impact jetty.  

But I guess it should not be considered an issue if it works for a standard Ubuntu.

**Comment by gsick** [comment]
Have you some error on /var/log/messages?

**Comment by jopasserat** [comment]
This file is not present on my system
